### PR TITLE
Add inplace kwarg to `torch.nn.functional.silu`

### LINKED ISSUE #455: Unexpected keyword arg 'inplace' for torch.nn.SiLU
### 🐛 Bug

NeMo's StableDiffusion uses [SiLU](https://pytorch.org/docs/stable/generated/torch.nn.SiLU.html) activation in its UNet model. When `UNet` stage is run with `thunder.jit`, thunder gives an error:
`silu() got an unexpected keyword argument 'inplace'`.
Specifically the error is is raised in this [line](https://github.com/pytorch/pytorch/blob/v2.3.0/torch/nn/modules/activation.py#L396) of `torch/nn/modules/activation.py` although `inplace` is a valid argument of the op.

CC: @tfogal 

cc @tfogal

### PR DESCRIPTION
**Pull request by riccardofelluga**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

Temporarily fixes #455 till we implement inplace operators by adding the keyword arg to the operator implementation.
I changed from testing the clang to testing the ltorch symbol as it is a superset of clang in this case so we test both.

### CODE REVIEW

File: thunder/torch/__init__.py [hunk 1]
```diff
@@ -1486,7 +1486,8 @@ def selu(a: TensorProxy, /, inplace: bool = False) -> TensorLike:
 
 
 @torchsymbol(torch.nn.functional.silu)
-def silu(a, /):
+def silu(a: TensorLike, inplace: bool = False):
+    utils.check(not inplace, lambda: "silu only supports inplace=False", exception_type=NotImplementedError)
```
Comment (t-vi) [inline thunder/torch/__init__.py:1490]: I might be nice to include "Thunder" somewhere in the error message so people know it's us not them...

**Review by t-vi — APPROVED** [review]
Thank you @riccardofelluga

**Comment by crcrpar** [comment]
why is `inplace` here kwarg unlike `inplace` of relu?

ref: https://github.com/Lightning-AI/lightning-thunder/blob/8309fc015c1ea9d684337cdcf95138524cd8645c/thunder/torch/__init__.py#L1459

**Comment by t-vi** [comment]
Probably because it's terrible to pass this without keyword, but you're right, we should ideally match the PyTorch signature.

**Pull request merged**